sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java

```diff
@@ -574,11 +573,7 @@ public DataSource apply(Void input) {
     @Override
     public PCollection<T> expand(PBegin input) {
       checkArgument(getQuery() != null, "withQuery() is required");
-      // getWithBeamSchema() is true then getBeamSchemaRowMapper() must not be null
-      checkArgument((getWithBeamSchema() == null || !getWithBeamSchema()) || getBeamSchemaRowMapper() != null, "withBeamSchemaRowMapper() is required");
-      // if getWithBeamSchema() is null or false then getRowMapper() must not be null
-      checkArgument((getWithBeamSchema() != null && getWithBeamSchema() ) || getRowMapper() != null, "withRowMapper() is required");
-//      checkArgument((getRowMapper() != null || (getWithBeamSchema() != null && getWithBeamSchema() && getBeamSchemaRowMapper() != null)), "withRowMapper() is required");
+      checkArgument(getBeamSchemaRowMapper() != null || getRowMapper() != null, "withRowMapper() is required if not using JdbcIO.readRows()");
```

Comment:
Would be easier to read if it was written like

```java
getBeamSchemaRowMapper() == null && getRowMapper() == null
```

Reply:
@charithe please have a look at the latest commit. We have removed BeamSchemaRowMapper and utilized RowMapper only.